aurpublish.in

```diff
@@ -22,6 +22,8 @@ usage() {
 		                      and a second copy of all commits in the subtree.
 		                      For more details, see the "--rejoin" option in git
 		                      subtree.
+		    -u, --url       Specify the URL of the server that should be used
+		                      for git operations.
```

Comment:
If for some reason people have an enduring need to regularly push to a non-default remote, it might be nice to have a way to specify this.

One way to do that without adding any new config files, even, is to piggyback off of git-config.

```
git config --get --local aur.remote-url
```

Which would allow persistently setting this. Do you think it would be valuable to support that?

Reply:
Yes, I could see that being a valuable feature for people who are self-hosting their own AUR. Though I do not do such a thing, so I am trying to imagine a possible self-hosting workflow in my head.

A couple thoughts/questions:

1. **Is `--local` beneficial in this case?**

    In my mind, this config capability only benefits users who rarely (or never) use the official AUR. If I try to imagine myself in this scenario then I am sure that I would quickly forget that I set a repo-specific git config option. So it might make more sense to allow this option to be set in the global config. That way they would not target the official AUR without explicitly setting the `--url` option - even on a fresh pull.

2. **Rather than "aur.remote-url" for the key, I would suggest "aurpublish.remoteUrl"**
    - Since this configuration option will be specific to this script, I think that it makes sense to use the script name as the topic key.
    - The [git config options](https://git-scm.com/docs/git-config) all seem to use camel case for option names. So it would probably be a good idea to match that pattern.

Reply:
It has been a couple weeks, so I went ahead with change using the suggestions that I proposed. I skipped `--local` option and used `aurpublish.remoteUrl` as the git config key.

See the latest version for implementation. If you disagree with my alterations please let me know.